pkg/csi/cinder/openstack/openstack.go

```diff
@@ -167,7 +196,17 @@ func CreateOpenStackProvider() (IOpenStack, error) {
 		provider.HTTPClient.Transport = netutil.SetOldTransportDefaults(&http.Transport{TLSClientConfig: config})
 	}
 
-	err = openstack.Authenticate(provider, authOpts)
+	if cfg.Global.TrustID != "" {
+		opts := cfg.toAuth3Options()
+		authOptsExt := trusts.AuthOptsExt{
+			TrustID:            cfg.Global.TrustID,
+			AuthOptionsBuilder: &opts,
```

Comment:
we can reuse cfg.toAuthOptions() here 
opts := cfg.toAuthOptions()

Reply:
just to be sure it identity v3 only ?

Reply:
then you can change it to use only v3 , cfg.toAuth3Options() ?  I dont see the need for maintaining both . wdyt?

Reply:
I'll check if this is working without the tokensv3 import, I'll let you know

Reply:
I think some people may not have the rights endpoints enabled in their openstack, or using old openstack versions where identity v3 is not yet present.
Thats 3 lines, so not a lot to maintain

Reply:
i think using the same bit of code found https://ghosthub.corp.blizzard.net/kubernetes/cloud-provider-openstack/blob/import/master/pkg/cloudprovider/providers/openstack/openstack.go#L413
until we can converge on a library might be the way to go. which it looks like it was OP's intent

Reply:
Sorry your link cannot be seen outside blizzard apparently, could you paste the lines ?

Reply:
oh whoops, meant to link upstream 

> i think using the same bit of code found https://github.com/kubernetes/cloud-provider-openstack/blob/master/pkg/cloudprovider/providers/openstack/openstack.go#L411
until we can converge on a library might be the way to go. which it looks like it was OP's intent